pkg/parser/parser.go

```diff
@@ -333,33 +346,60 @@ func (p *Parser) createServiceTasks(protoTasks []*api.Task) ([]*api.Task, error)
 			return nil, fmt.Errorf("%w: %v", ErrInternal, err)
 		}
 
+		// Craft Docker build arguments
+		var buildArgs string
+		for key, value := range taskContainer.DockerArguments {
+			buildArgs += fmt.Sprintf(" %s=%s", key, value)
+		}
+
 		newTask := &api.Task{
 			Name:         fmt.Sprintf("Prebuild %s", taskContainer.DockerfilePath),
 			LocalGroupId: p.NextTaskID(),
 			Instance:     anyInstance,
 			Commands: []*api.Command{
 				{
-					Name: "dummy",
+					Name: "build",
+					Instruction: &api.Command_ScriptInstruction{
+						ScriptInstruction: &api.ScriptInstruction{
+							Scripts: []string{fmt.Sprintf("docker build "+
+								"--tag gcr.io/%s:%s "+
+								"--file %s%s "+
+								"${CIRRUS_DOCKER_CONTEXT:-$CIRRUS_WORKING_DIR}",
+								prebuiltInstance.Repository, prebuiltInstance.Reference,
+								buildArgs, taskContainer.DockerfilePath)},
+						},
+					},
+				},
+				{
+					Name: "push",
 					Instruction: &api.Command_ScriptInstruction{
 						ScriptInstruction: &api.ScriptInstruction{
-							Scripts: []string{"true"},
+							Scripts: []string{fmt.Sprintf("gcloud docker -- push gcr.io/cirrus-ci-community/%s:latest",
+								dockerfileHash)},
 						},
 					},
 				},
 			},
+			Environment: map[string]string{
+				"DISPLAY": ":99.0",
```

Comment:
I don't think it's there by default.

Reply:
Indeed 🤦‍♂️

Fixed in https://github.com/cirruslabs/cirrus-cli/pull/113/commits/cc8755b55854fb4a51c1c5e37ba4ed04a26aef5d.